support/supportedversion/version_test.go

```diff
@@ -198,3 +207,202 @@ func TestGetMinSupportedVersion(t *testing.T) {
+		expectErr             bool
+		expectedErrMsg        string
+		expectedVersions      SupportedVersions
+		expectedServerVersion string
+	}{
+		{
+			name: "When the ConfigMap is valid, expect versions to be returned successfully",
+			cm: &corev1.ConfigMap{
+				ObjectMeta: baseCM.ObjectMeta,
+				Data: map[string]string{
+					config.ConfigMapVersionsKey:      string(validVersionsJSON),
+					config.ConfigMapServerVersionKey: "test-server-version",
+				},
+			},
+			expectErr:             false,
+			expectedVersions:      validVersions,
+			expectedServerVersion: "test-server-version",
+		},
+		{
+			name:           "When the ConfigMap is not found, expect an error",
+			cm:             nil, // No configmap will be added to the client
+			expectErr:      true,
+			expectedErrMsg: "failed to find supported versions on the server",
+		},
+		{
+			name: "When the server-version key is missing, expect an error",
+			cm: &corev1.ConfigMap{
+				ObjectMeta: baseCM.ObjectMeta,
+				Data:       map[string]string{config.ConfigMapVersionsKey: string(validVersionsJSON)},
+			},
+			expectErr:      true,
+			expectedErrMsg: "the server did not advertise its HyperShift version",
+		},
+		{
+			name: "When the supported-versions key is missing, expect an error",
+			cm: &corev1.ConfigMap{
+				ObjectMeta: baseCM.ObjectMeta,
+				Data:       map[string]string{config.ConfigMapServerVersionKey: "test-server-version"},
+			},
+			expectErr:      true,
+			expectedErrMsg: "the server did not advertise supported OCP versions",
+		},
+		{
+			name: "When the supported-versions JSON is malformed, expect an error",
+			cm: &corev1.ConfigMap{
+				ObjectMeta: baseCM.ObjectMeta,
+				Data: map[string]string{
+					config.ConfigMapVersionsKey:      `{"versions": "not-an-array"}`,
+					config.ConfigMapServerVersionKey: "test-server-version",
+				},
+			},
+			expectErr:      true,
+			expectedErrMsg: "failed to parse supported versions on the server",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewGomegaWithT(t)
+
+			// Setup fake client
+			scheme := api.Scheme
+			g.Expect(corev1.AddToScheme(scheme)).To(Succeed())
+			var fakeClient client.Client
+			if tc.cm != nil {
+				fakeClient = fake.NewClientBuilder().WithScheme(scheme).WithObjects(tc.cm).Build()
+			} else {
+				fakeClient = fake.NewClientBuilder().WithScheme(scheme).Build()
+			}
+
+			// Execute the function
+			supportedVersions, serverVersion, err := GetSupportedOCPVersions(context.Background(), namespace, fakeClient)
+
+			// Assert results
+			if tc.expectErr {
+				g.Expect(err).To(HaveOccurred())
+				g.Expect(err.Error()).To(ContainSubstring(tc.expectedErrMsg))
+			} else {
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(supportedVersions).To(Equal(tc.expectedVersions))
+				g.Expect(serverVersion).To(Equal(tc.expectedServerVersion))
+			}
+		})
+	}
+}
+
+func TestRetrieveSupportedOCPVersion(t *testing.T) {
```

Comment:
can we please not introduce unit tests which does http requests? this needs to be mocked

Reply:
Updated